lib/cc/yaml/nodes/engine.rb

```diff
@@ -4,7 +4,28 @@ module Nodes
       class Engine < Mapping
         map :enabled, to: Scalar[:bool], required: true
         map :checks, to: Checks
-        map :config, to: EngineConfig
+        map :config, to: Mapping
+
+        attr_reader :config
+
+        def visit_key_value(visitor, key, value)
+          if key == "config"
+            raw_value = value.to_ruby
```

Comment:
The full YAML implementation allows for encoding of arbitrary Ruby objects, which is unsafe. I don't think we can use regular Psych to_ruby safely.

In a car right now but will need to look at this further today